File: apps/ios/ScribbleFit/ScribbleFit/Features/AI/Domain/Models/AnalysisSummary.swift
import Foundation

public struct AnalysisSummary: Sendable {
    public let period: SummaryPeriod
    public let summaryText: String
    public let highlights: [String]
    public let muscleDistribution: [MuscleGroupStat]
    public let focusArea: String
    public let volumeDelta: Double
    public let timestamp: Date
    
    public init(
        period: SummaryPeriod,
        summaryText: String,
        highlights: [String],
        muscleDistribution: [MuscleGroupStat],
        focusArea: String,
        volumeDelta: Double,
        timestamp: Date
    ) {
        self.period = period
        self.summaryText = summaryText
        self.highlights = highlights
        self.muscleDistribution = muscleDistribution
        self.focusArea = focusArea
        self.volumeDelta = volumeDelta
        self.timestamp = timestamp
    }
}

public struct MuscleGroupStat: Sendable {
    public let muscleGroup: String
    public let volumePercentage: Double
    
    public init(muscleGroup: String, volumePercentage: Double) {
        self.muscleGroup = muscleGroup
        self.volumePercentage = volumePercentage
    }
}

public enum SummaryPeriod: String, Codable, Sendable {
    case day, week, month, year
}

File: apps/android/feature/ai/domain/src/main/java/com/scribblefit/feature/ai/domain/model/AnalysisSummary.kt
package com.scribblefit.feature.ai.domain.model

import kotlinx.serialization.Serializable

@Serializable
data class AnalysisSummary(
    val period: SummaryPeriod,
    val summaryText: String,
    val highlights: List<String>,
    val muscleDistribution: List<MuscleGroupStat>,
    val focusArea: String,
    val volumeDelta: Double,
    val timestamp: Long
)

@Serializable
data class MuscleGroupStat(val muscleGroup: String, val volumePercentage: Double)

@Serializable
enum class SummaryPeriod { DAY, WEEK, MONTH, YEAR }
